config/initializers/doorkeeper.rb

```diff
@@ -10,7 +10,7 @@
   # This block will be called to check whether the resource owner is authenticated or not.
   resource_owner_authenticator do
     @user ||= User.from_auth(cookies.signed[:auth])
-    unless @user
+    if @user.blank? || !@user.terms_of_service
```

Comment:
Could you add a test for this?

Crucially, what should happen if the user hasn't accepted the terms of service? There should be some sort of error that is included in the API response, right?